packages/files-ui/src/Components/Modules/FileBrowsers/Sharing/LinkList.tsx

```diff
@@ -234,47 +193,30 @@ const LinkList = ({ bucketId, bucketEncryptionKey }: Props) => {
       )}
       {nonces.length < MAX_LINKS && (
         <>
-          <Typography
-            component="h4"
-            variant="h4"
-            className={classes.heading}
-          >
-            <Trans>Create a sharing link</Trans>
-          </Typography>
-          <div className={classes.grayWrapper}>
-            <div className={classes.creationWrapper}>
-              <div className={classes.rightSelection}>
-                <Typography
-                  component="h5"
-                  variant="h5"
-                  className={classes.rightsText}
-                >
-                  <Trans>Anyone with the link can: </Trans>
-                </Typography>
-                <MenuDropdown
-                  title={(newLinkPermission && translatedPermission(newLinkPermission)) || ""}
-                  anchor="bottom-right"
-                  className={classes.permissionDropdown}
-                  classNames={{
-                    icon: classes.icon,
-                    options: classes.options,
-                    title: classes.dropdownTitle
-                  }}
-                  testId="permission"
-                  menuItems={displayedItems}
-                />
-              </div>
-              <Button
-                className={classes.createLinkButton}
-                onClick={onCreateNonce}
-                variant="secondary"
-                disabled={isLoadingCreation}
-                loading={isLoadingCreation}
-                data-cy="button-create-link"
-              >
-                <Trans>Create link</Trans>
-              </Button>
-            </div>
+          <div className={classes.generateLinkContainer}>
+            <Button
+              className={classes.createLinkButton}
+              onClick={onCreateNonce}
+              variant="link"
+              disabled={isLoadingCreation}
+              loading={isLoadingCreation}
+              data-cy="button-create-link"
+            >
+              <LinkIcon className={classes.linkIcon} />
+              <Trans>Generate sharing link</Trans>
+            </Button>
+            <PermissionsDropdown
+              selectedPermission={newLinkPermission || "read"}
+              permissions={nonces.length === 0 ? ["read", "write"] : hasAReadNonce ? ["write"] : ["read"]}
+              onViewPermissionClick={() => setNewLinkPermission("read")}
+              onEditPermissionClick={() => setNewLinkPermission("write")}
+              withBorders={false}
+              injectedClasses={{
+                options: classes.options,
+                dropdownTitle: classes.dropdownTitle
+              }}
+              testId="link-permission-dropdown"
```

Comment:
We don't need to use "dropdown" here as it will be constructed with a "dropdown-" prefix when sent as a testId parameter (set up in `MenuDropdown.tsx`). 

The convention for identifiers that I've been following is to put the element type at the beginning. 

```suggestion
              testId="link-permission"
```